src/taipy/core/submission/submission.py

```diff
@@ -0,0 +1,180 @@
+        version (Optional[str]): The string indicates the application version of the submission to instantiate.
+            If not provided, the latest version is used.
+    """
+
+    _ID_PREFIX = "SUBMISSION"
+    _MANAGER_NAME = "submission"
+    __SEPARATOR = "_"
+
+    def __init__(
+        self,
+        entity_id: str,
+        id: Optional[str] = None,
+        jobs: Optional[Union[List[Job], List[JobId]]] = None,
+        creation_date: Optional[datetime] = None,
+        submission_status: Optional[SubmissionStatus] = None,
+        version: Optional[str] = None,
+    ):
+        self._entity_id = entity_id
+        self.id = id or self.__new_id()
+        self._jobs: Union[List[Job], List[JobId], List] = jobs or []
+        self._creation_date = creation_date or datetime.now()
+        self._submission_status = submission_status or SubmissionStatus.SUBMITTED
+        self._version = version or _VersionManagerFactory._build_manager()._get_latest_version()
+
+    @staticmethod
+    def __new_id() -> str:
+        """Generate a unique Submission identifier."""
+        return SubmissionId(Submission.__SEPARATOR.join([Submission._ID_PREFIX, str(uuid.uuid4())]))
+
+    @property
+    def entity_id(self) -> str:
+        return self._entity_id
+
+    @property
+    def creation_date(self):
+        return self._creation_date
+
+    def get_label(self) -> str:
+        """Returns the submission simple label prefixed by its owner label.
+
+        Returns:
+            The label of the submission as a string.
+        """
+        return self._get_label()
+
+    def get_simple_label(self) -> str:
+        """Returns the submission simple label.
+
+        Returns:
+            The simple label of the submission as a string.
+        """
+        return self._get_simple_label()
+
+    @property  # type: ignore
+    @_self_reload(_MANAGER_NAME)
+    def jobs(self) -> List[Job]:
+        jobs = []
+        job_manager = _JobManagerFactory._build_manager()
+
+        for job in self._jobs:
+            jobs.append(job_manager._get(job))
+
+        return jobs
+
+    @jobs.setter  # type: ignore
+    @_self_setter(_MANAGER_NAME)
+    def jobs(self, jobs: Union[List[Job], List[JobId]]):
+        self._jobs = jobs
+
+    def __hash__(self):
+        return hash(self.id)
+
+    def __eq__(self, other):
+        return self.id == other.id
+
+    @property  # type: ignore
+    @_self_reload(_MANAGER_NAME)
+    def submission_status(self):
+        return self._submission_status
+
+    @submission_status.setter  # type: ignore
+    @_self_setter(_MANAGER_NAME)
+    def submission_status(self, submission_status):
+        self._submission_status = submission_status
+
+    def __lt__(self, other):
+        return self.creation_date.timestamp() < other.creation_date.timestamp()
+
+    def __le__(self, other):
+        return self.creation_date.timestamp() == other.creation_date.timestamp() or self < other
```

Comment:
🥜 Why not simply `return self.creation_date.timestamp() <= other.creation_date.timestamp()`?

Reply:
thanks for pointing this out! I noticed the same issue on Job, will be fixing that as well!